src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -442,31 +450,100 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
             return null;
         }
 
+        private bool IsParamsArrayExpanded(SemanticModel semanticModel, SyntaxNode node, SymbolInfo symbolInfo, CancellationToken cancellationToken)
+        {
+            if (symbolInfo.Symbol == null)
+            {
+                return false;
+            }
+
+            int argumentCount;
+            bool lastArgumentHasNameColon;
+            ExpressionSyntax lastArgumentExpression;
+
+            if (node is AttributeSyntax attribute)
+            {
+                argumentCount = attribute.ArgumentList.Arguments.Count;
+                lastArgumentHasNameColon = attribute.ArgumentList.Arguments.LastOrDefault()?.NameColon != null ||
+                    attribute.ArgumentList.Arguments.LastOrDefault()?.NameEquals != null;
+                lastArgumentExpression = attribute.ArgumentList.Arguments.LastOrDefault()?.Expression;
+            }
+            else
+            {
+                BaseArgumentListSyntax argumentList = node switch
+                {
+                    InvocationExpressionSyntax invocation => invocation.ArgumentList,
+                    ObjectCreationExpressionSyntax objectCreation => objectCreation.ArgumentList,
+                    ConstructorInitializerSyntax constructorInitializer => constructorInitializer.ArgumentList,
+                    ElementAccessExpressionSyntax elementAccess => elementAccess.ArgumentList,
+                    _ => throw new ArgumentException("Unexpected SyntaxNode", nameof(node))
+                };
+
+                argumentCount = argumentList.Arguments.Count;
+                lastArgumentHasNameColon = argumentList.Arguments.LastOrDefault()?.NameColon != null;
+                lastArgumentExpression = argumentList.Arguments.LastOrDefault()?.Expression;
+            }
+
+
+            if (symbolInfo.Symbol is IMethodSymbol methodSymbol && methodSymbol.Parameters.LastOrDefault()?.IsParams == true)
+            {
+                if (argumentCount > ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                {
+                    return true;
+                }
+                else if (argumentCount == ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                {
+                    if (lastArgumentHasNameColon)
+                    {
+                        return true;
```

Comment:
can you comment (or give example) to explain why this is true?  i'm not really getting it.

Reply:
You're not getting it because it was wrong. Switched to false and added a test. 😬